Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java

```diff
@@ -183,25 +171,39 @@ public void collectSignal(final MaxAdapterSignalCollectionParameters parameters,
     @Override
     public void loadInterstitialAd(final MaxAdapterResponseParameters parameters, final Activity activity, final MaxInterstitialAdapterListener listener)
     {
+        String appId = parameters.getServerParameters().getString( "app_id", null );
+        if ( TextUtils.isEmpty( appId ) )
+        {
+            log( "No app ID provided" );
+            listener.onInterstitialAdLoadFailed( MaxAdapterError.INVALID_CONFIGURATION );
+            return;
+        }
+
         String bidResponse = parameters.getBidResponse();
         boolean isBiddingAd = AppLovinSdkUtils.isValidString( bidResponse );
         String placementId = parameters.getThirdPartyAdPlacementId();
+        Context context = getContext( activity );
         log( "Loading " + ( isBiddingAd ? "bidding " : "" ) + "interstitial ad for placement: " + placementId + "..." );
 
-        if ( !VungleAds.isInitialized() )
-        {
-            log( "Vungle SDK not successfully initialized: failing interstitial ad load..." );
-            listener.onInterstitialAdLoadFailed( MaxAdapterError.NOT_INITIALIZED );
-
-            return;
-        }
-
         updateUserPrivacySettings( parameters );
 
-        interstitialAd = new InterstitialAd( getContext( activity ), placementId, new AdConfig() );
-        interstitialAd.setAdListener( new InterstitialListener( listener ) );
+        VungleInitializer.getInstance().initialize( appId, context, new VungleInitializationListener()
```

Comment:
@felixzhang0703 @HuFangshuai question here is should we just fail the ad and then init the SDK so that sdk will be initialized for the second load. 

[Admob does](https://github.com/googleads/googleads-mobile-ios-mediation/blob/main/adapters/LiftoffMonetize/LiftoffMonetizeAdapter/GADMAdapterVungleInterstitial.m#L90) implementation is fail the ad and init the sdk.

Reply:
admob does not fail the ad

Reply:
i didn't check the max ios implementation. maybe you mean that instead?

Reply:
sorry my bad. resolving this.